HHS/Head-Start-TTADP
# [TTAHUB-871] Add AR to AWS Elasticsearch Index on Save

## Description of change

When a report is submitted we want to create a searchable document in the 'activityreports' index. If the report is reset to draft and changes are made we want to update this document with the latest values. If the report is deleted we want to remove this document from the index.

Note: This code will only create the report document when the report is 'Submitted'.

## How to test

With both Docker and they Worker running (yarn worker).

1. Run the 'yarn createAwsElasticSearchIndexes' 
2. Create a new report and submit
3. Check both the Docker and Worker terminals for messages (pass/fail)
4. Reset the report back to Draft and make changes
5. Check both the Docker and Worker terminals for messages (pass/fail)
6. Reset the Draft and Delete the report
7. Check both the Docker and Worker terminals for messages (pass/fail)

With a newly report submitted you can query the docker image directly to ensure your document is created:

**curl -X GET http://localhost:9200/activityreports/_search**

Then check that you updates show with the same query. 

Deleting the report from the UI should remove the report document from the query results.

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-871


## Checklists

### Every PR



### Production Deploy


### After merge/deploy
